api/apps/sessions/model.js

```diff
@@ -0,0 +1,55 @@
+import Joi from 'api/lib/joi.coffee'
+import db from 'api/lib/db.coffee'
+import { omit } from 'lodash'
+import { ObjectId } from 'mongojs'
+
+const schema = Joi.object().keys({
+  timestamp: Joi.date(),
+  user: Joi.object().keys({
+    id: Joi.string().objectid(),
+    name: Joi.string()
+  }),
+  article_id: Joi.string().objectid(),
+  channel_id: Joi.string().objectid()
+})
+
+const querySchema = {
+  q: Joi.string().allow('')
+}
+
+export const where = (input, callback) => {
+  Joi.validate(input, querySchema, (err, input) => {
+    if (err) {
+      return callback(err)
+    }
+
+    const cursor = db
+      .sessions
+      .find(input)
+
+    cursor.toArray((err, sessions) => {
+      callback(err, sessions)
+    })
+  })
+}
+
+export const save = (input, callback) => {
+  Joi.validate(input, schema, {}, (err, input) => {
+    if (err) {
+      callback(err)
+    }
+
+    const data = {
+      ...omit(input, 'id'),
+      _id: input.id
+    }
+
+    db.sessions.save(data, callback)
```

Comment:
I'm getting the following error here:

```
POST /api/sessions 403 10.307 ms - 104
TypeError: Cannot read property 'save' of undefined
  at /Users/lucsucces/Projects/positron/api/apps/sessions/model.js:84:26
  at internals.Object._validateWithOptions (/Users/lucsucces/Projects/positron/node_modules/joi/lib/types/any/index.js:654:20)
  at module.exports.internals.Any.root.validate (/Users/lucsucces/Projects/positron/node_modules/joi/lib/index.js:121:23)
  at save (/Users/lucsucces/Projects/positron/api/apps/sessions/model.js:75:17)
  at saveSession (/Users/lucsucces/Projects/positron/api/apps/sessions/routes.js:42:17)
  at Layer.handle [as handle_request] (/Users/lucsucces/Projects/positron/node_modules/express/lib/router/layer.js:95:5)
  at next (/Users/lucsucces/Projects/positron/node_modules/express/lib/router/route.js:137:13)
  at Route.dispatch (/Users/lucsucces/Projects/positron/node_modules/express/lib/router/route.js:112:3)
```
Do I need to create collection manually? @kanaabe